nibabel/spatialimages.py

```diff
@@ -324,6 +325,43 @@ class ImageDataError(Exception):
 class SpatialImage(DataobjImage):
     ''' Template class for volumetric (3D/4D) images '''
     header_class = SpatialHeader
+    # Make a strong assumption that the first three dimensions are spatial
+    # Subclasses/images for which this is not true should change this attribute
+    _spatial_dims = slice(0, 3)
+
+    class Slicer(object):
+        ''' Slicing interface that returns a new image with an updated affine
+        '''
+        def __init__(self, img):
+            self.img = img
+
+        def __getitem__(self, slicer):
+            klass = self.img.__class__
+            if not klass.makeable:
+                raise NotImplementedError(
+                    "Cannot slice un-makeable image types")
+
+            try:
+                slicer = self.img._check_slicing(self._arr_to_slice(slicer))
+            except ValueError as err:
+                raise IndexError(*err.args)
```

Comment:
`canonical_slicers` raises `ValueError`s on things that would be `IndexError`s in numpy. Since they're being used in a `__getitem__` context, I make the conversion here so the expected exception type is raised without changing the API elsewhere.